### PR TITLE
Refactor: Move global staticFilters constant to class method

### PR DESCRIPTION
**Pull request by d-alleyne**
Follow-up to https://github.com/Automattic/wp-calypso/pull/95995#pullrequestreview-2412589439

## Proposed Changes

- Refactor `staticFilters` from a globally scoped constant to a method `getStaticFilters` within the ThemeShowcase component.
- Use the `props.translate()` function provided by the `localize()` HOC to ensure translation accuracy based on the specific i18n context.
- Ensure filter text dynamically updates in response to translation context changes, aligning with user session translation data.

## Why are these changes being made?

These changes address feedback from a [previous code review](https://github.com/Automattic/wp-calypso/pull/95995#pullrequestreview-2412589439), which highlighted the limitations of using static `translate()` calls. By moving the filter's definition inside the `ThemeShowcase` component, we leverage the props.translate() method, reflecting the localized translation instance tied to the user's session, which may differ from the global one. This ensures filters display accurately across different translation contexts, particularly in environments like `/<locale>/themes` where previously, static translations appeared in English due to data discrepancies between global and session-specific translation instances.

## Testing Instructions

- Switch the user locale to Mag-16 locale, e.g. Spanish

- Visit `/home` for a website

- Hover over `Appearance` and wait for the Popup menu to appear with the `Themes` option
image

- Click the `Themes` (Temas in Spanish) and observe that `My Themes` filter menu remains translated.
image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/themes/theme-showcase.jsx [hunk 1]
```diff
@@ -177,7 +152,34 @@ class ThemeShowcase extends Component {
 
 	isThemeDiscoveryEnabled = () => config.isEnabled( 'themes/discovery' );
 
-	getDefaultStaticFilter = () => defaultStaticFilter;
+	translatedStaticFilters() {
+		const { translate } = this.props;
+		return {
+			MYTHEMES: {
+				key: STATIC_FILTERS.MYTHEMES,
+				get text() {
```
Comment (yuliyan) [inline client/my-sites/themes/theme-showcase.jsx]: I suppose the getters are no longer needed.

File: client/my-sites/themes/theme-showcase.jsx [hunk 2]
```diff
@@ -177,7 +152,34 @@ class ThemeShowcase extends Component {
 
 	isThemeDiscoveryEnabled = () => config.isEnabled( 'themes/discovery' );
 
-	getDefaultStaticFilter = () => defaultStaticFilter;
+	translatedStaticFilters() {
```
Comment (yuliyan) [inline client/my-sites/themes/theme-showcase.jsx]: Nit: Having "translated" in the method name seems a bit redundant.

**Review by yuliyan — APPROVED** [review]
Everything seems to work as expected and SSR filters are now translated.

File: client/my-sites/themes/theme-showcase.jsx (hunk 2 above)
Comment (d-alleyne) [reply client/my-sites/themes/theme-showcase.jsx]: I needed a name to differentiate it from the `staticFilters` constant that was being defined within the methods.

**Pull request merged**